homeassistant/components/kitchen_sink/lock.py

```diff
@@ -0,0 +1,92 @@
+"""Demo platform that has a couple of fake locks."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.lock import LockEntity, LockEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_LOCKED, STATE_UNLOCKED, STATE_UNLOCKING
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Demo sensors."""
+    async_add_entities(
+        [
+            DemoLock(
+                "kitchen_sink_lock_001",
+                "Openable kitchen sink lock",
```

Comment:
Maybe this name is a bit weird, it sounds like it's a lock for a kitchen sink.
How about just this:
```suggestion
                "Openable lock",
```

Reply:
I'll address this in a follow up PR

Reply:
#86452